src/storage/mvcc/reader/scanner/backward.rs

```diff
@@ -227,15 +240,39 @@ impl<S: Snapshot> BackwardKvScanner<S> {
         }
         assert!(ts > last_checked_commit_ts);
 
+        // Check newer data if needed
+        let mut use_near_seek = false;
+        if self.cfg.check_can_be_cached && self.can_be_cached {
+            let seek_key = user_key.clone().append_ts(TimeStamp::max());
+            self.write_cursor
+                .internal_seek(&seek_key, &mut self.statistics.write)?;
+            use_near_seek = true;
+            assert!(self.write_cursor.valid()?);
+            let current_key = self.write_cursor.key(&mut self.statistics.write);
+            assert!(Key::is_user_key_eq(
+                current_key,
+                user_key.as_encoded().as_slice()
+            ));
+            let max_ts = Key::decode_ts_from(current_key)?;
+            if max_ts > ts {
+                self.can_be_cached = false
+            }
+        }
+
         // After several `prev()`, we still not get the latest version for the specified ts,
         // use seek to locate the latest version.
```

Comment:
maybe we should put these comments before L243.

Reply:
Addressed